hls4ml/model/optimizer/passes/repack_stream.py

```diff
@@ -53,3 +86,38 @@ def transform(self, model, node):
         model.replace_node(node, repack_layer)
 
         return True
+
+class BroadcastStream(OptimizerPass):
+    def match(self, node):
+        if node.__class__.__name__ == 'Merge':
+            try: 
+                inp1 = node.get_input_variable(node.inputs[0])
+                inp2 = node.get_input_variable(node.inputs[1])
+                return inp1.shape != inp2.shape
+            except:
+                return False
+        else:
+            return False
+        
+    def transform(self, model, node):
+        if model.config.backend.name not in ['Vivado'] or \
+            model.config.get_config_value('IOType') != 'io_stream':
```

Comment:
Doesn't this also affect io_parallel?

Reply:
So I believe this transformation is skipped if IOType is io_parallel given the above condition (i.e. it returns False immediately if IOType is io_parallel). Basically, the above condition is equivalent to:
```python
if not (model.config.backend.name == 'Vivado' and model.config.get_config_value('IOType') == 'io_stream'): return False
```

Are you saying you want it to also be applied in the case of io_parallel? That would require some additional modifications, and is currently planned to be done in a follow-up PR to generalize this broadcasting to more cases + io_parallel and io_stream.

Reply:
Sorry for my poor choice of words, I was indeed asking if the broadcasting issue affects `io_parallel` as well, which I thought it does. I agree that support for it should not be a part of this PR and a follow-up PR is a better way forward.